eth/stagedsync/stage_headers.go

```diff
@@ -137,8 +137,19 @@ func HeadersPOS(
 ) error {
 	// Waiting for the beacon chain
 	log.Info("Waiting for payloads...")
+	var payloadMessage privateapi.PayloadMessage
 	atomic.StoreUint32(cfg.waitingPosHeaders, 1)
-	payloadMessage := <-cfg.reverseDownloadCh
+	// Decide what kind of action we need to take place
+	select {
+	case payloadMessage = <-cfg.reverseDownloadCh:
+	case <-cfg.hd.SkipCycleHack:
+		atomic.StoreUint32(cfg.waitingPosHeaders, 0)
+		if !useExternalTx {
```

Comment:
There's probably no need to commit the transaction since we're at the very beginning of the method and haven't written anything to the DB yet.